src/viam/sdk/robot/client.hpp

```diff
@@ -138,10 +138,7 @@ class RobotClient {
     /// @throws `std::runtime_error` if the requested resource doesn't exist or is the wrong type.
     /// @return a `shared_ptr` to the requested resource.
     std::shared_ptr<T> resource_by_name(std::string name) {
-        API api = API::get<T>();
-        Name r = Name(api, "", name);
-
-        auto resource = this->resource_by_name(std::move(r));
+        auto resource = this->resource_by_name({API::get<T>(), "", name});
```

Comment:
I don't think you need the `this->` here. And the function is now simple enough it could be a one liner:

```
std::shared_ptr<T> resource_by_name(std::string name) {
    return std::dynamic_pointer_cast<T>(resource_by_name({API::get<T>(), "", std::move(name)}));
}
```

Note that `name` should be moved from.